### PR TITLE
BankIdP + some docker stuff

### PR DESCRIPTION
**Pull request by theseal**

### CODE REVIEW

File: manifests/docker_compose.pp
Comment (mickenordin) [inline manifests/docker_compose.pp:1]: This one I am not sure about... I would suggest something like:

```
  $nftenabled_and_interface =  $::facts['sunet_nftables_enabled'] == 'yes' and has_key($::facts['networking']['interfaces'], 'to_docker')
  $advanced_network_or_nftdisabled = $::facts['dockerhost_advanced_network'] == 'yes' or $::facts['sunet_nftables_enabled'] == 'no'
  if ( $nftenabled_and_interface or  $advanced_network_or_nftdisabled ) {
    $_install_service = true
  } else {
    $_install_service = false
    notice("sunet::docker_compose: Not installing ${service_name}, interface to_docker missing")
  }

  if $_install_service {
```

**Review by mickenordin — CHANGES_REQUESTED** [review]
I revied the stuff about advanced network and didn't check the bankidp stuff :)

File: manifests/docker_compose.pp
Comment (mikaelfrykholm) [reply manifests/docker_compose.pp:1]: Yes, I struggled with that logic as well. I failed when I tried re-setting variables.

Comment (mikaelfrykholm) [reply manifests/docker_compose.pp:1]: your suggestion worked out of the box

**Review by mickenordin — APPROVED** [review]
Looks good!

File: manifests/rediscluster.pp [hunk 1]
```diff
@@ -29,9 +32,11 @@
       ensure  => present,
       content => template('sunet/rediscluster/server.conf.erb'),
     }
-    -> sunet::misc::ufw_allow { "redis_port_${i}":
```
Comment (mickenordin) [inline manifests/rediscluster.pp:32]: Should this be wrapped in:

```
if $::facts['sunet_nftables_enabled'] == 'yes'  or $::facts['dockerhost_advanced_network'] == 'yes' {
....
}
```

Leaving the old ufw rules inplace if those conditions are not met?

**Review by mickenordin — CHANGES_REQUESTED** [review]

File: manifests/rediscluster.pp (hunk 1 above)
Comment (mikaelfrykholm) [reply manifests/rediscluster.pp:32]: Good catch, fixed!

**Review by mickenordin — APPROVED** [review]
lgtm

**Pull request merged**